.github/workflows/docs.yml

```diff
@@ -22,69 +21,68 @@ on:
       - pyproject.toml
 
 jobs:
-  Docs:
-    name: linux docs
-    runs-on: ubuntu-latest
-    env:
-      DOC_VERSION: dev
+  test-docs:
     defaults:
       run:
         shell: bash -l {0}
-    concurrency:
-      group: ${{ github.workflow }}-${{ github.ref }}
-      cancel-in-progress: true
-    timeout-minutes: 15
+    env:
+      DOC_PYTHON_VERSION: '3.11'
+    runs-on: ubuntu-latest
     steps:
       - name: checkout code
         uses: actions/checkout@v3
 
-      - name: Setup Mambaforge
-        uses: conda-incubator/setup-miniconda@v2
-        with:
-            python-version: 3.11
-            miniforge-variant: Mambaforge
-            miniforge-version: latest
-            channels: conda-forge
-            activate-environment: hydromt
-            use-mamba: true
+      # we need environment.yml to see if we have a cache hit
       - name: Generate env spec
-        run: python make_env.py doc
-
-
-      - name: Set cache date
-        run: echo "DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+        run: pip install tomli && python make_env.py test,io,extra,doc -p ${{ env.DOC_PYTHON_VERSION }} -n hydromt
 
-      - name: load from cache if it exists for our environment
-        uses: actions/cache@v3
-        with:
-          path: /usr/share/miniconda3/envs/hydromt
-          key: docs-conda-${{ hashFiles('environment.yml') }}-${{ env.DATE }}
+      - name: load from cache
         id: cache
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            /usr/share/miniconda3
+            ~/pycache
+            ~/docs/_build
+          # the below two settings mean we'll alway srestore the cache
+          # but the cache hit output can tell us if we have to update afterwards
+          key: docs-${{ hashFiles('environment.yml')  }}
+          restore-keys: |
+            docs
 
-      - name: Update environment & write to cache
-        run: mamba env update -n hydromt -f environment.yml
-        if: steps.cache.outputs.cache-hit != 'true'
+      - name: Fail on no cache restore
```

Comment:
Does this mean the workflow fails if there is no cache available? What would be the expected cases for when no caching is available?

Reply:
it does, this should actually never happen, but because mamba itself is in the cache the flow would fail anyway, this is just to provide a bit of a nicer error message. In the case the cache is unavailable, it should be regenerated (which happens on a schedule)